libs/vectorise/include/vectorise/fixed_point/fixed_point.hpp

```diff
@@ -697,12 +749,16 @@ template <typename T>
 constexpr FixedPoint<I, F>::FixedPoint(T n, meta::IfIsFloat<T> * /*unused*/)
   : data_(static_cast<typename FixedPoint<I, F>::Type>(n * ONE_MASK))
 {
-  if (CheckOverflow(static_cast<NextType>(n) * static_cast<NextType>(ONE_MASK)))
+  if (IsMinusZero(n))
+  {
+    data_ = 0;
```

Comment:
Do you ignore the sign?

Reply:
-0.0 is the same as +0.0, however in fixed-point comparing to -0.0 means comparing to a big integer, which leads to an unwanted underflow. Hence we need to detect and remove -0.0 value.

Reply:
> -0.0 is the same as +0.0
I remember reading a text years ago where they were apparently distinct, something about modelling something in gas dynamics, sorry, can't seem to retrieve it now.

Reply:
in terms of FixedPoint numbers, they are definitely not distinct. The problem is that while the FPU does that in hardware, and -0.0 is 0x80000000 (for 32-bit floats), you can do things like -1.0 * 0.0 and still get 0x80000000. However in our Int class, if we allow negative zero, we will have as a result of that operation, 0xffffffffffffffffffffffffffffffff, which causes an underflow. Since negative zero has no usefulness in our case, I opted to remove support for it entirely. You are probably refering  to some physical problems which use a limit to zero from the negative x. In floating point types this may have an application as the exponents are quite large compared to a fixed point, but even with Int<256> we don't reach such exponents.